### PR TITLE
Missing warnings for @inlined attributes in classic mode

### PR DESCRIPTION
**Pull request by lthls**
(@Keryan-dev please review when you have some time -- I may have missed a few cases)

### CODE REVIEW

**Comment by lthls** [comment]
Note: Closure has a distinction between unknown functions and functions with no information, but to do it in classic mode we would need a bit of refactoring so I've left it out.
There's no warning for ignored `@inline` attributes, but this matches Closure.

**Review by Keryan-dev — COMMENTED** [review]
Looks fine. I put a CR somewhere saying warnings are missing that can be decommissioned.

**Review by Keryan-dev — APPROVED** [review]

**Comment by mshinwell** [comment]
This isn't quite right yet, as discussed

**Comment by mshinwell** [comment]
@lthls what's the status of this one?

**Comment by lthls** [comment]
I don't remember exactly what we discussed. This PR should get warnings in the right places, but I'm not sure we'll consistently get the exact same warnings as either Closure or Simplify. So if there are tests that check the exact warning string (and I suspect there are in the testsuite), we might fail those in classic mode.

**Comment by mshinwell** [comment]
@lthls maybe we should rebase and merge this?  If the warnings are slightly different it should be ok, and we can iterate anyway.

File: middle_end/flambda2/simplify_shared/inlining_helpers.ml [hunk 1]
```diff
@@ -115,3 +115,12 @@ let wrap_inlined_body_for_exn_extra_args acc ~extra_args ~apply_exn_continuation
   in
   let_cont_create acc wrapper ~handler_params:wrapper_handler_params
     ~handler:wrapper_handler ~body:body_with_push ~is_exn_handler:true
+
+type attribute_kind =
+  | Inlined
+  | Unroll
+
+let string_of_kind = function Inlined -> "[@inlined]" | Unroll -> "[@unroll]]"
```
Comment (Gbury) [inline middle_end/flambda2/simplify_shared/inlining_helpers.ml:123]:
```suggestion
let string_of_kind = function Inlined -> "[@inlined]" | Unroll -> "[@unrolled]]"
```

Only the `[@unrolled <n>]` annotation exists (cf the flambda manual : https://v2.ocaml.org/manual/flambda.html#s%3Aflambda-manual-control ).

File: middle_end/flambda2/simplify_shared/inlining_helpers.ml [hunk 2]
```diff
@@ -115,3 +115,12 @@ let wrap_inlined_body_for_exn_extra_args acc ~extra_args ~apply_exn_continuation
   in
   let_cont_create acc wrapper ~handler_params:wrapper_handler_params
     ~handler:wrapper_handler ~body:body_with_push ~is_exn_handler:true
+
+type attribute_kind =
+  | Inlined
+  | Unroll
```
Comment (Gbury) [inline middle_end/flambda2/simplify_shared/inlining_helpers.ml:121]:
```suggestion
  | Unrolled
```

As said in another comment, we only have "unrolled", not "unroll"

**Review by Gbury — CHANGES_REQUESTED** [review]
This looks good (modulo the unroll/unrolled comment), and it adds a warning on partial applications which is currently missing, but for partial applications, we should also remove the attribute, else Simplify will emit another warning about unknown function information (which is extremely confusing).

**Comment by lthls** [comment]
I've fixed the names and tweaked the partial application stub to remove the inlining attributes from the original application.
I think that issue #1261 will be fixed by this PR (at least the error messages on the sample program look coherent).

**Pull request merged**